include/openmc/particle.h

```diff
@@ -181,6 +181,15 @@ class Particle {
     int delayed_group; //!< particle delayed group
   };
 
+  //! Saved ("banked") state of primary source particle, for particle restart
+  struct RestartBank {
+    Position r;
+    Direction u;
+    double E;
+    double wgt;
+  };
```

Comment:
A thought here -- if we save the state of the RNG stream at the time the source was sampled, we should be able to reproduce the state of the particle at the beginning of it's lifetime, no? If so, that might be a better option than adding more data to the (already bloated) Particle class.

Reply:
Suggested changes have been applied. The data has been removed from the particle class and for fixed source the sampling is done from the distribution when writing the particle restart data. For eigenvalue simulations the data is obtained form the primary source bank as was the original method